php-checks/src/main/java/org/sonar/php/checks/security/RequestContentLengthCheck.java

```diff
@@ -0,0 +1,124 @@
+@Rule(key = "S5693")
+public class RequestContentLengthCheck extends PHPVisitorCheck {
+  public static final int DEFAULT = 8_000_000;
+  @RuleProperty(
+    key = "fileUploadSizeLimit",
+    defaultValue = "" + DEFAULT)
+  long fileUploadSizeLimit = DEFAULT;
+
+  private static final QualifiedName SYMFONY_FILE_CONSTRAINT = qualifiedName("Symfony\\Component\\Validator\\Constraints\\File");
+  private static final Pattern SIZE_FORMAT = Pattern.compile("^([0-9]+)(k|M|Ki|Mi)?$");
+
+  private static final String MESSAGE = "Make sure the content length limit is safe here.";
+
+  @Override
+  public void visitNewExpression(NewExpressionTree tree) {
+    if (isInstantiationOf(tree, SYMFONY_FILE_CONSTRAINT) && ((FunctionCallTree)tree.expression()).callArguments().size() == 1) {
+      checkSymfonyFileConstraint(((FunctionCallTree)tree.expression()).callArguments().get(0).value());
+    }
+    super.visitNewExpression(tree);
+  }
+
+  private void checkSymfonyFileConstraint(ExpressionTree tree) {
+    ExpressionTree value = CheckUtils.assignedValue(tree);
+
+    if (!value.is(Tree.Kind.ARRAY_INITIALIZER_BRACKET, Tree.Kind.ARRAY_INITIALIZER_FUNCTION)) {
+      return;
+    }
+
+    Optional<ExpressionTree> setSize = arrayValue((ArrayInitializerTree) value, "maxSize");
+    if (setSize.isPresent()) {
+      checkFileSize(setSize.get());
+    } else {
+      context().newIssue(this, value, MESSAGE);
+    }
+  }
+
+  private void checkFileSize(ExpressionTree size) {
+    ExpressionTree sizeValue = CheckUtils.assignedValue(size);
+    long setBytes = 0;
+
+    if (sizeValue.is(Tree.Kind.NUMERIC_LITERAL, Tree.Kind.REGULAR_STRING_LITERAL)) {
+      Matcher matcher = SIZE_FORMAT.matcher(sizeValue.is(Tree.Kind.REGULAR_STRING_LITERAL) ? CheckUtils.trimQuotes((LiteralTree) sizeValue) : ((LiteralTree)sizeValue).value());
+      if (!matcher.matches()) {
+        return;
+      }
+
+      setBytes = Long.parseLong(matcher.group(1));
+
+      String unit = matcher.group(2) != null ? matcher.group(2) : "";
+      switch (unit) {
+        case "k":
+          setBytes *= 1000;
+          break;
+        case "M":
+          setBytes *= 1_000_000;
+          break;
+        case "Ki":
+          setBytes *= 1024;
+          break;
+        case "Mi":
+          setBytes *= 1_048_576;
+          break;
+        default:
+      }
+    } else if (sizeValue.is(Tree.Kind.NULL_LITERAL)) {
+      // null means unlimited
+      setBytes = fileUploadSizeLimit + 1;
+    }
+
+    if (setBytes > fileUploadSizeLimit) {
+      context().newIssue(this, sizeValue, MESSAGE);
+    }
+  }
+
+  private boolean isInstantiationOf(NewExpressionTree tree, QualifiedName name) {
+    ExpressionTree expression = tree.expression();
+    if (!expression.is(Tree.Kind.FUNCTION_CALL) || !((FunctionCallTree)expression).callee().is(Tree.Kind.NAMESPACE_NAME)) {
+      return false;
+    }
+
+    return name.equals(getFullyQualifiedName((NamespaceNameTree) ((FunctionCallTree) expression).callee()));
+  }
```

Comment:
Second time I see this today. Maybe yo can move this to CheckUtils.

Reply:
I see no elegant solution to avoid this currently. We would also need to pass the symbol table to that method and repeat what `PHPVisitorCheck#getFullyQualifiedName` is doing. Let's refactor this separately.